File: pubsub/src/main/java/glitch/pubsub/object/json/VideoPlayback.java
package glitch.pubsub.object.json;

import com.google.gson.annotations.JsonAdapter;
import glitch.pubsub.object.adapters.VideoPlaybackTypeAdapter;
import lombok.Data;

import java.time.Instant;

@Data
public class VideoPlayback {
    @JsonAdapter(VideoPlaybackTypeAdapter.class)
    private final Type type;
    private final Instant serverTime;

    public enum Type {
        STREAM_UP,
        STREAM_DOWN,
        VIEW_COUNT
    }
}

File: pubsub/src/main/kotlin/glitch/pubsub/object/json/VideoPlayback.kt
package glitch.pubsub.`object`.json

import com.google.gson.annotations.JsonAdapter
import glitch.pubsub.`object`.adapters.VideoPlaybackTypeAdapter
import java.time.Instant

class VideoPlayback(
        @JsonAdapter(VideoPlaybackTypeAdapter::class)
        val type: Type, val serverTime: Instant) {

    enum class Type {
        STREAM_UP,
        STREAM_DOWN,
        VIEW_COUNT
    }
}
